Sources/ToolchainRegistry/ToolchainRegistry.swift

```diff
@@ -283,7 +286,18 @@ package final actor ToolchainRegistry {
   /// If we have a toolchain in the toolchain registry that contains the compiler with the given URL, return it.
   /// Otherwise, return `nil`.
   package func toolchain(withCompiler compiler: URL) -> Toolchain? {
-    return toolchainsByCompiler[compiler]
+    if let resolvedPath = try? compiler.realpath {
+      return toolchainsByCompiler[resolvedPath]
+    }
+    return nil
+  }
+
+  /// If we have a toolchain in the toolchain registry with the given URL, return it. Otherwise, return `nil`.
+  package func toolchain(withPath path: URL) -> Toolchain? {
+    if let resolvedPath = try? path.realpath {
+      return toolchainsByPath[resolvedPath]
+    }
+    return nil
```

Comment:
Similar to my comment on `toolchain(withCompiler:)`, should we try a non-realpath lookup first and then cache if necessary to avoid hitting the file system unnecessarily? E.g. something like

```swift
if let toolchain = toolchainsByPath[path] {
  return toolchain
}
guard let realpath = orLog("Realpath toolchain", { try path.realpath }) else {
  return nil
}
guard let toolchain = lchainsByPath[realpath] else {
  return nil
}
// Cache mapping of non-realpath to the realpath toolchain for faster subsequent lookups
toolchainsByPath[path] = toolchain
return toolchain
```